src/js/controller/controller.js

```diff
@@ -662,18 +664,25 @@ Object.assign(Controller.prototype, {
 
         function _nextUp() {
             const related = _api.getPlugin('related');
-            if (related) {
-                const nextUp = _model.get('nextUp');
-                if (nextUp) {
-                    _this.trigger('nextClick', {
-                        mode: nextUp.mode,
-                        ui: 'nextup',
-                        target: nextUp,
-                        itemsShown: [ nextUp ],
-                        feedData: nextUp.feedData,
-                    });
-                }
-                related.next();
+            triggerAdvanceEvent(related, 'nextClick', () => related.next());
+        }
+
+        function triggerAdvanceEvent(related, evt, cb) {
+            if (!related) {
```

Comment:
I think it makes more sense to not have `related` as an argument because you're never using `related` at your callsites, outside of passing it to this function.

Reply:
I agree. I passed it in the avoid doing the plugin lookup twice (the callback for `nextClick` requires a reference to the related plugin). Another option is passing the related plugin into the callback and that way we keep `related` specific to this function but that's a little hard to follow when reading through the code imo. How do you feel about just doing the lookup twice in the case of the `nextclick` handler?

Reply:
Oh I see what you mean actually. Fine the way it is.

Reply:
You're not calling it twice, but it's repeated twice, instead of only being written and called once in `triggerAdvanceEvent`.

Reply:
oh i see. passing in `() => related.next()`